src/lustre/lustreAbstractInterpretation.ml

```diff
@@ -260,9 +268,15 @@ and interpret_eqn node_id ctx ty_ctx lhs rhs =
       | LA.SingleIdent (_, id) ->
         let ty1 = Ctx.lookup_ty ty_ctx id |> get in
         let ty1 = Ctx.expand_nested_type_syn ty_ctx ty1 in
-        let ty2 = interpret_expr_by_type node_id ctx ty_ctx ty1 p expr in
-        let ty = restrict_type_by ty1 ty2 in
-        add_type acc node_id id ty
+        let restrict_ty = interpret_expr_by_type node_id ctx ty_ctx ty1 p expr in
+        let ty, is_restricted = restrict_type_by ty1 restrict_ty in
+        Format.eprintf "id: %a@. ty1: %a@. restrict_ty: %a@."
+          HString.pp_print_hstring id
+          LA.pp_print_lustre_type ty1
+          LA.pp_print_lustre_type restrict_ty;
```

Comment:
You forgot to disable this debugging message :)

Reply:
Oops